### PR TITLE
Feature/2016 07 25 web of trusted commits

### PR DESCRIPTION
**Pull request by kweem**

### CODE REVIEW

File: _posts/2016-07-25-Web-of-trusted-commits.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+layout: post
+authors: [martin_kwee]
+title: 'A web of trusted commits'
+image: /img/digitally-signing-your-json-documents.png
+tags: [Security, Git, Encryption, Digital Signature, Cryptography]
+category: Security
+comments: true
+---
+## Who Do You Trust?
+When you’re building software with people from around the world, it’s important to validate that commits and tags are coming from an identified source. By using a distributed revision control system like Git, anyone can have an offline copy of your project's code repository. In theory having a central repository is not necessary, but it can be used to provide an "official" source from which other developers can clone from and work on. These other floating repositories may contain malicious code because it is remarkably easy to fake your identity when committing code using Git.
```
Comment (bart-blommaerts) [inline _posts/2016-07-25-Web-of-trusted-commits.md]: because, unfortunately, ?

File: _posts/2016-07-25-Web-of-trusted-commits.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
++ **Signing**
+  - The message is still readable to everyone.
+  - Guarantee of the sender's identity (aka Sherlock).
+  - Guarantee that the message has not been tampered with since it has been signed by the sender (aka Sherlock).
+
++ **Encryption**
+  - The message is only readable by the designated recipient (aka John).
+  - No guarantee of the sender's identity (aka Sherlock).
+  - Encryption can be done symmetrically by using a Shared Secret Key, a single key is then used for both encryption and decryption. Asymmetrical encryption (aka Public Key encryption) with a Public/Private Keypair uses one key for encryption and another for decryption. Note that the advantages and challenges of using either encryption type is beyond the scope of this blog post.
+
```
Comment (bart-blommaerts) [inline _posts/2016-07-25-Web-of-trusted-commits.md]: Add nice drawing of pub/priv key usage?

File: _posts/2016-07-25-Web-of-trusted-commits.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+## Enforcing Trust
+
+Sherlock will combine a digital signature with encryption to convince John that his message is trustworthy.
+
+1. Sherlock wants to send the following message to John: `Data! Data! Data! I can’t make bricks without clay.`. He calculates the **Hash** of this message by applying a publicly known hashing algorithm (e.g. MD5, SHA-256, ...) to the message. The calculated hash by using the SHA-256 algorithm is `d6ba26816599a75310c4c263126d4b44979c7026f90e1db8e9b317d6658f3811`. The hash value is unique to the hashed data.
+
```
Comment (bart-blommaerts) [inline _posts/2016-07-25-Web-of-trusted-commits.md]: Might not be a good a idea to mention MD5? https://en.wikipedia.org/wiki/MD5

File: _posts/2016-07-25-Web-of-trusted-commits.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      {% endhighlight %}
+
++ Specify the key you want, or accept the default `RSA and RSA`. RSA is a widely-used asymmetric encryption algorithm and is named after Ron Rivest, Adi Shamir and Len Adleman who invented it in 1977. Should you be interested in more mathematical details how this algorithm works, I can highly recommend watching ["Public Key Cryptography: RSA Encryption Algorithm"](https://www.youtube.com/watch?v=wXB-V_Keiu8) on YouTube.
+
++ Enter the desired key size. I recommend the `maximum key size of 4096 bits` because they provide far better long-term security. While the default of 2048 bits is secure now, it won't be in the future. 1024 bit keys are already considered within the range of being breakable and while technology advances 2048 bit keys will also become breakable. Eventually 4096 bit keys will be broken too, but that will be so far in the future that better encryption algorithms will also likely have been developed by then.
+
++ You can safely accept the `default expiration` for your key.
+
++ Enter `your real name and email address`. If you have a GitHub account, then you should enter the verified email address for your account.
+
```
Comment (bart-blommaerts) [inline _posts/2016-07-25-Web-of-trusted-commits.md]: You? Or Sherlock? :)

File: _posts/2016-07-25-Web-of-trusted-commits.md (hunk 2 above)
Comment (kencoenen) [inline _posts/2016-07-25-Web-of-trusted-commits.md]: You can also give examples of both encryption types. I think that the Belgian eID uses asymmetrical encryption.

Maybe also highlight symmetrical en asymmetrical to stress that there are two types?

**Comment by bart-blommaerts** [comment]
Really like this style .. Looking forward to Sherlock's new adventures :)

Maybe, add a small "commercial" paragraph at the end: Since Ordina is a secure-by-design company, We advise all developers to use this way of working ... But in your own words. Pretty sure, management will appreciate :)

@TYsewyn Your take on this?

**Pull request merged**